### PR TITLE
Manual OAuth 2.0

### PR DESCRIPTION
**Pull request by ryanwi**
Implement the OAuth flow manually since the `client-oauth2` library doesn't support [PKCE for OAuth 2.0](https://oauth.net/2/pkce/)

### CODE REVIEW

**Comment by ryanwi** [comment]
Note, we've also been exploring using auth.js https://github.com/signalwire/call-fabric-client-beta/compare/joao/use_auth_js?expand=1

for the example/reference, maybe we want to show it more than rely on the lib, I dunno?

File: index.js [hunk 1]
```diff
@@ -71,21 +64,56 @@ app.get('/minimal', async (req, res) => {
...
+  // console.log("Request Headers: ", req.headers);
+
+  const params = new URLSearchParams();
+  params.append('client_id', process.env.OAUTH_CLIENT_ID);
+  params.append('grant_type', 'authorization_code');
+  params.append('code', req.query.code);
+  params.append('redirect_uri', process.env.OAUTH_REDIRECT_URI);
+  params.append('code_verifier', req.session.verifier);
+
+  const response = await fetch(process.env.OAUTH_TOKEN_URI, {
```
Comment (iAmmar7) [inline index.js:99]: We have `axios` installed as well. I think for consistency, we should stick to one. Maybe we can remove the axios.

**Review by iAmmar7 — APPROVED** [review]
LGTM, just a few minor suggestions.

**Review by jpsantosbh — APPROVED** [review]
just consider my comment before merging it

File: index.js (hunk 1 above)
Comment (ryanwi) [reply index.js:97]: Yeah, seeing that auth.js relies on fetch made we wonder why we use axios still, I can convert the other method and remove axios if we think removing it is fine.

**Comment by ryanwi** [comment]
Good points, this is public and will be likely shared with clients I think.

That said, the oauth standard makes the client implementation fairly straightforward, it's not a custom auth scheme.

We can start with this maybe and then continue to explore auth.js more?

**Pull request merged**